### ISSUE TITLE
SSH broken on fish shell

### ISSUE THREAD

**Issue by marcellourbani** [warpdotdev/Warp#4986]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

Fish raises an error as its command line is incompatible with warp's wrapper:
```
ssh myservrobert

fish: Unsupported use of '='. In fish, please use 'set hook $(printf "{\"hook\": \"SSH\", \"value\": {\"socket_path\": \"~/.ssh/31507\", \"remote_shell\": \"%s\"}}" "${SHELL##*/}" | command od -An -v -tx1 | command tr -d " \n")'.
hook=$(printf "{\"hook\": \"SSH\", \"value\": {\"socket_path\": \"~/.ssh/31507\", \"remote_shell\": \"%s\"}}" "${SHELL##*/}" | command od -An -v -tx1 | command tr -d " \n")
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
Connection to myservrobert.bti.local closed.
```

### To reproduce

1. start fish shell in a warp terminal
2. try to ssh anywhere

### Expected behavior

my remote system prompt

### Screenshots

![image](https://github.com/warpdotdev/Warp/assets/2453277/4c20264b-03a6-4bb9-ab53-7be08a083376)

### Operating system

Linux

### Operating system and version

Arch

### Shell Version

fish 3.7.1

### Current Warp version

v0.2024.05.28.08.02.stable_00

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Additional context

works fine if I start it from bash

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Warp has an SSH Wrapper that enables features (like the blocks, prompt, completions, etc.) on remote machines requiring the default prompt on the remote machine to be either `bash` or `zsh` (not `fish` at this time).

As a workaround you can add `command ssh` to your `Settings > Subshells > Added commands` menu, then run `command ssh <user@server>` to connect to remote session, this will attempt to enable Warp features as a subshell and work with `fish`.

You can also try disabling the SSH Wrapper by navigating to `Settings > Features`. Blocks will stop working but with it disabling, SSH should work as normal.

Related docs:
[https://docs.warp.dev/features/ssh](https://docs.warp.dev/features/ssh#ssh-wrapper-fails)
https://docs.warp.dev/features/subshells

If those don't work, please let us know.

**Comment by superm1** [comment]
I was about to make a new issue, but I found this one looking for dupes.  It's continuing to happen with WARP `v0.2025.01.22.08.02.stable_02` and fish `3.7.1` .

**Comment by m0xsec** [comment]
> I was about to make a new issue, but I found this one looking for dupes. It's continuing to happen with WARP `v0.2025.01.22.08.02.stable_02` and fish `3.7.1` .
> 
> Warp seems to advertise it _should_ work though.
> 
> ![Image](https://github.com/user-attachments/assets/0dc37038-3c37-4f98-a0a8-b7c19ca5e3e5)

yeah i am confused by this also - hopefully the replacement SSH subshell stuff is in the near future

**Comment by GuillemCForgeFlow** [comment]
@dannyneira Could I kindly ask about if this is something that may be fixed into the future? I can also see there is a related one #871 which attracted quite some attention.

Thank you in advance 🙂

**Comment by regalstreak** [comment]
any updates here? still breaking

**Comment by hongfeiyang** [comment]
same error, fish 4.7.1, warp v0.2026.05.18.05.32.stable_02
